pkg/policy/api/l4.go

```diff
@@ -73,3 +73,17 @@ type L7Rules struct {
 	// +optional
 	Kafka []PortRuleKafka `json:"kafka,omitempty"`
 }
+
+// Len returns the total number of rules inside `L7Rules`.
+// Returns 0 if nil.
+func (rules *L7Rules) Len() int {
```

Comment:
Unit test these functions please. Do the existing policy resolution unit tests cover these cases? If not, please add such a test.